### PR TITLE
rocm_set_soversion added 

### PR DESCRIPTION
**Pull request by saadrahim**
Usable by math libraries to manage the ABI version MAJOR parsing.

### CODE REVIEW

File: share/rocm/cmake/ROCMSetupVersion.cmake [hunk 1]
```diff
@@ -154,3 +154,12 @@ function(rocm_setup_version)
     endif()
 
 endfunction()
+
+function(rocm_set_soversion)
+    set(oneValueArgs LIBRARY_TARGET SOVERSION)
+
+    cmake_parse_arguments(PARSE "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
```
Comment (pfultz2) [inline share/rocm/cmake/ROCMSetupVersion.cmake]: There is actually no need to use `cmake_parse_arguments`. I think doing `function(rocm_set_soversion LIBRARY_TARGET SOVERSION)` would work and should be clear enough when called(ie `rocm_set_soversion(rocfft 0.1)`

Comment (saadrahim) [reply share/rocm/cmake/ROCMSetupVersion.cmake]: Without the parse, both variables were not usable in the function.

Comment (saadrahim) [reply share/rocm/cmake/ROCMSetupVersion.cmake]: I had to add the parse after trying without

Comment (saadrahim) [reply share/rocm/cmake/ROCMSetupVersion.cmake]: I see what you are asking, fixing it.

**Pull request merged**